ClassTranscribeServer/Controllers/MediaController.cs

```diff
@@ -182,12 +182,15 @@ public async Task<ActionResult<Media>> PostMedia(IFormFile video1, IFormFile vid
                 await video1.CopyToAsync(stream);
             }
 
-            // Only do this for the first (primary) video
-            media.UniqueMediaIdentifier = await FileRecord.ComputeSha256HashForFileAsync(filePath);
-
             media.JsonMetadata.Add("video1", JsonConvert.SerializeObject(video1));
             media.JsonMetadata.Add("video1Path", filePath);
-            video.Video1 = await FileRecord.GetNewFileRecordAsync(filePath, Path.GetExtension(filePath));
+
+            var playlist = await _context.Playlists.FindAsync(playlistId);
+            var co = CommonUtils.GetRelatedCourseOffering(playlist);
+            video.Video1 = await FileRecord.GetNewFileRecordAsync(filePath, Path.GetExtension(filePath), co);
+
+            // Only do this for the first (primary) video
+            media.UniqueMediaIdentifier = video.Video1.Hash;
```

Comment:
Is video1.Hash exactly equivalent to FileRecord.ComputeSha256HashForFileAsync(filePath); ?Othewise do we download it again, next time we do a sync?
"Hash" property *might* be better as a method, ComputeHash() - it is not obvious that this is an expensive operation

Reply:
Yes, `video1.Hash` is exactly equivalent to `FileRecord.ComputeSha256HashForFileAsync(filePath)`. In the `GetNewFileRecordAsync` method, the `fileRecord.Hash` field is set to `ComputeSha256HashForFileAsync(fileRecord.Path)`, and then that `fileRecord` object is returned. So I just changed this so we're not calculating the same hash two times in a row.